testutils/clustermanager/gke.go

```diff
@@ -299,65 +222,39 @@ func (gc *GKECluster) Acquire() error {
 	for i, region := range regions {
 		// Restore innocence
 		err = nil
-		rb := &container.CreateClusterRequest{
-			Cluster: &container.Cluster{
-				NodePools: []*container.NodePool{
-					{
-						Name: "default-pool",
-						Autoscaling: &container.NodePoolAutoscaling{
-							Enabled:      true,
-							MinNodeCount: gc.Request.MinNodes,
-							MaxNodeCount: gc.Request.MaxNodes,
-						},
-						Config: &container.NodeConfig{
-							MachineType: gc.Request.NodeType,
-						},
-					},
-				},
-				Name: clusterName,
-				// The default cluster version is not latest, has to explicitly
-				// set it as "latest"
-				InitialClusterVersion: defaultGKEVersion,
-				// Installing addons after cluster creation takes at least 5
-				// minutes, so install addons as part of cluster creation, which
-				// doesn't seem to add much time on top of cluster creation
-				AddonsConfig: gc.getAddonsConfig(),
-				// Equivalent to --enable-basic-auth, so that user:pass can be
-				// later on retrieved for setting up cluster roles. Use the
-				// default username from gcloud command, the password will be
-				// automatically generated by GKE SDK
-				MasterAuth: &container.MasterAuth{Username: "admin"},
-			},
-			ProjectId: *gc.Project,
+		var rb *container.CreateClusterRequest
+		rb, err = gke.NewCreateClusterRequest(&gc.Request.Request)
```

Comment:
is it possible to create using`Request` directly? Also, do you have to do `Request.Request`? Can you passing `GKERequest` straight as `Request`?

Reply:
There will be a reference cycle if I pass `GKERequest`..
But let me think if there is a better way to handle it.

Reply:
Automatically solved after adding the deepcopy :-)